students/Wayne/Assignments/Lesson5/Mailroom2.py

```diff
@@ -0,0 +1,256 @@
+"""
+Simple mailroom program for a non profit
+"""
+
+# Start with interactive loop
+
+import sys
+import math
+
+from textwrap import dedent
+
+
+"""
+The following section of code pertain to the building and manipulation of the
+donors database.
+"""
+
+# ----------------------------------------------------------------------------
+#
+# Built a sample database and put it into a dict format
+#
+# ----------------------------------------------------------------------------
+
+
+def find_donor_db():
+    return{'dave grohl': ("Dave Grohl", [999999.99, 45.63]),
+           'billy joe armstrong': ("Billy Joe Armstrong", [15000, 2222]),
+           'rivers cuomo': ("Rivers Cuomo", [1994.96, 2100000]),
+           }
+
+# ----------------------------------------------------------------------------
+#
+# The following function adds a new donor to the db. The function below uses
+# .lower() to convert all of the donor names into a single case to allow for
+# an easier sorting method of the dict.
+#
+# ----------------------------------------------------------------------------
+
+
+def add_new_donor(name):
+    name = name.strip()
+    donor = (name, [])
+    donor_db[name.lower()] = donor
+    return donor
+
+# ----------------------------------------------------------------------------
+#
+# The following function uses the dictionary key to search the donor database
+#
+# ----------------------------------------------------------------------------
+
+
+def search_donor_db(name):
+    key = name.strip().lower()
+    return donor_db.get(key)
+
+# ----------------------------------------------------------------------------
+#
+# The following function creates a list of donors from the database
+#
+# ----------------------------------------------------------------------------
+
+
+def list_donors():
+    listing = ("Donors:")
+    for donor in donor_db.values():
+        listing.append(donor[0])
+    return "\n".join(listing)
+
+
+# ----------------------------------------------------------------------------
+#
+# The following function adds a new donation amount into the database
+#
+# ----------------------------------------------------------------------------
+
+
+def accept_donation(name):
+    while True:
```

Comment:
Why loop here? You are not repeating at all. You are either returning or breaking out of the loop, so this loop doesn't seem necessary.